.github/actions/cache-go-dependencies/action.yaml

```diff
@@ -25,5 +25,10 @@ runs:
         key: go-build-v1-${{ github.job }}-${{ hashFiles('**/go.sum') }}
 
     - name: Download Go modules
+      env:
+        # > When GOTOOLCHAIN is set to local, the go command always runs the bundled Go toolchain.
+        # To prevent accidental updates of Go.
+        # We need to use `local` to not break downstream builds.
+        GOTOOLCHAIN: local
```

Comment:
Thinking of a better place to put this override, this target is likely invoked from `make deps` call.
https://github.com/stackrox/stackrox/blob/badb3f6e699ed52901fefc8217e48e4413cb7090/Makefile#L345-L356

It already has a conditional for `CI`. It can be similarly extended to change GOTOOLCHAIN when in CI context.

Reply:
OK, I was not sure if we should modify makefile for this but as it will only run in CI I think it's fine and should not break local workflow